File: data/data_transpiled/at_coder/abc017/C/3874429.py
def import import sys , math , out
import os
import sys , math
class Main ( object ) :
    mod = 1000000007
    def __init__ ( self ) :
        self.mod = mod
    def run ( self ) :
        with open ( self.infile , 'r' ) as f :
            n , m = struct.unpack ( '>i' , f.read ( ) )
            score = [ 0 ] * ( m + 2 )
            sum = 0
            for l , r , s in [ ( l , r , s ) for s in [ 1 , 2 , 3 , 4 ] ] :
                score [ l ] += s
                score [ r + 1 ] -= s
                sum += s
            for i in range ( 1 , m + 1 ) :
                score [ i ] += score [ i - 1 ]
        ans = 0
        for i in range ( 1 , m + 1 ) :
            ans = max ( ans , sum - score [ i ] )
        out.write ( ans )

File: data/data_original/at_coder/abc017/C/3874429.java
import java.util.*;
import static java.lang.System.*;
import static java.lang.Math.*;
public class Main {
    static int mod=1000000007;
    public static void main(String[]$) {
        Scanner sc = new Scanner(in);
        //??:???????i?????????????????score[]???????????????????????
        //??i????????????i???????????????0????????????????????????????
        int n=sc.nextInt();
        int m=sc.nextInt();
        long[] score=new long[m+2];
        long sum=0;
        for (int i = 0; i <n; i++) {
            int l=sc.nextInt(),r=sc.nextInt(),s=sc.nextInt();
            score[l]+=s;
            score[r+1]-=s;
            sum+=s;
        }
        for (int i = 1; i <=m; i++) {
            score[i]+=score[i-1];
        }
        long ans=0;
        for (int i = 1; i <=m; i++) {
            ans=max(ans,sum-score[i]);
        }
        out.println(ans);
    }
}